client/src/overview/overview.jsx

```diff
@@ -1,19 +1,39 @@
-import React from 'react';
+import React, { useContext, useState } from 'react';
+import Contexts from '../contexts/Contexts.tsx';
 import AddToCart from './subComponents/AddToCart.jsx';
 import ImageGallery from './subComponents/ImageGallery.jsx';
-import ProductInfo from './subComponents/ProductInfo.jsx';
+import ProductInfo from './subComponents/ProductInfo/ProductInfo.jsx';
 import StyleSelector from './subComponents/StyleSelector.jsx';
+const StyleIdxContext = React.createContext();
+const StyleNameContext = React.createContext();
 
-export default function OverView(props) {
+function OverView() {
+  const currentProductStyle = useContext(Contexts.ProductStyleContext);
+  const stylePics = [];
+  const [currentStyleIdx, setCurrentStyleIdx] = useState(0);
+  const [currentStyleName, setCurrentStyleName] = useState('Forest Green & Black by Default');
+  if (currentProductStyle) {
+    currentProductStyle.results.forEach((element, index) => {
+      stylePics.push(element.photos[0].thumbnail_url);
+    });
+  }
   return (
     <div className="overViewContainer">
       <div className="overView">
-        {/* <h2>OverView!</h2> */}
-        <ImageGallery />
-        <ProductInfo />
-        <StyleSelector />
-        <AddToCart />
+        <StyleIdxContext.Provider value={{ currentStyleIdx, setCurrentStyleIdx }}>
+          <StyleNameContext.Provider value={{ currentStyleName, setCurrentStyleName }}>
+            <ImageGallery />
+            <ProductInfo />
+            <StyleSelector
+              stylePics={ stylePics.length ? stylePics : null }
```

Comment:
What is the purpose of passing `null` if the list has a length of 0? Passing `null` instead of `[]` seems like it might cause some unexpected errors down the line